client/engine/engine.go

```diff
@@ -130,6 +130,11 @@ func (e *Engine) handleMessage(message protocols.Message) ObjectiveChangeEvent {
 		e.logger.Print(err)
 		return ObjectiveChangeEvent{}
 	}
+	err = e.store.SetObjective(updatedObjective)
```

Comment:
Do we need to set the objective here? We are going to set it inside `attemptProgress` anyway?

I think its important we have a really well thought out scheme for when we read and write to the store.

Reply:
I think you're right and we can probably just rely on setting it in `attemptProgress`

Reply:
[b1a4096](https://github.com/statechannels/go-nitro/pull/274/commits/b1a40960b99c176f9acd1c5f3209d127ad3b0763)